src/components/wallet-manager/index.tsx

```diff
@@ -1,116 +1,132 @@
+      setIsConnecting(false);
     }
-  }
+  }, [closeModal, connectionStatus]);
 
-  get showButton() {
-    return !(
-      ( this.props.connectionStatus === ConnectionStatus.Connected ) &&
-      ( this.props.currentConnector === Connectors.Metamask )
+  useEffect(() => {
+    if (walletSelected || connectionStatus === ConnectionStatus.Connecting) {
+      setIsConnecting(true);
+    }
+  }, [connectionStatus, walletSelected]);
+
+  useEffect(() => {
+    setShowButton(
+      !(
+        connectionStatus === ConnectionStatus.Connected &&
+        currentConnector === Connectors.Metamask
+      )
     );
-  }
-
-  get showModal() {
-    return this.state.showModal;
-  }
-
-  get isConnecting() {
-    return this.state.walletSelected || ( this.props.connectionStatus === ConnectionStatus.Connecting );
-  }
-
-  get availableWallets() {
-    return [
-      WalletType.Metamask,
-      WalletType.WalletConnect,
-      WalletType.Coinbase,
-      WalletType.Fortmatic,
-      WalletType.Portis,
-    ];
-  }
-
-  openModal = () => this.setState({ showModal: true });
-  closeModal = () => this.setState({ showModal: false });
-
-  handleWalletSelected = (connector: WalletType) => {
-    this.setState({ walletSelected: true });
-    this.props.updateConnector(connector);
-  }
-
-  render() {
-    return (
-      <div className="wallet-manager">
-        {this.props.currentAddress && <EthAddress address={this.props.currentAddress} />}
-        {this.showButton && (
-          <Button
-            className='wallet-manager__connect-button'
-            label='Connect'
-            onClick={this.openModal}
-          />
-        )}
-        {this.showModal && (
+  }, [connectionStatus, currentConnector]);
+
+  const handleWalletSelected = useCallback(
+    (connector: WalletType): void => {
+      setWalletSelected(true);
+      updateConnector(connector);
+    },
+    [updateConnector]
+  );
+
+  return (
+    <div className='wallet-manager'>
+      {currentAddress && (
+        <div data-testid='eth-address'>
+          <EthAddress address={currentAddress} />
+        </div>
+      )}
+      {showButton && (
+        <Button
+          data-testid='connect-button'
+          className='wallet-manager__connect-button'
+          label='Connect'
+          onClick={openModal}
+        />
+      )}
+      {showModal && (
+        <div>
           <WalletSelectModal
-            wallets={this.availableWallets}
-            isConnecting={this.isConnecting}
-            onClose={this.closeModal}
-            onSelect={this.handleWalletSelected}
+            wallets={availableWallets}
+            isConnecting={isConnecting}
+            onClose={closeModal}
+            onSelect={handleWalletSelected}
           />
-        )}
-      </div>
-    );
-  }
-}
-
-export const WalletManager = connectContainer<{}>(Container);
+        </div>
+      )}
+    </div>
+  );
+};
+
+const mapStateToProps = (state: RootState): Partial<Props> => {
```

Comment:
What's the best way to test state mapping? It looks like [redux-mock-store](https://github.com/reduxjs/redux-mock-store) was an option. Is there a more modern alternative?